opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -422,6 +446,24 @@ def get_laser_envelope( self, t=None, iteration=None, pol=None, m='all',
         if index != 'all':
             info.restrict_to_1Daxis( info.axes[1] )
 
+        # Plot the result if needed
+        if plot:
+            iteration = self.iterations[ self.current_i ]
+            time_fs = 1.e15 * self.t[ self.current_i ]
+            if index != 'all':
+                plt.plot( info.z, envelope, **kw)
+                plt.ylabel('$E_%s \;(V/m)$' % pol,
```

Comment:
Actually, info.z is in SI (meters in this case). So, I would use `1.e6*info.z` here, since the labels indicate microns.